src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -985,5 +990,62 @@ private static IExpressionStatement CreateBoundExpressionStatementOperation(Boun
             Optional<object> constantValue = default(Optional<object>);
             return new LazyExpressionStatement(expression, isInvalid, syntax, type, constantValue);
         }
+
+        private static IInterpolatedStringExpression CreateBoundInterpolatedStringExpressionOperation(BoundInterpolatedString boundInterpolatedString)
+        {
+            Lazy<ImmutableArray<IInterpolatedStringContent>> parts = new Lazy<ImmutableArray<IInterpolatedStringContent>>(() => GetInterpolatedStringExpressionParts(boundInterpolatedString));
+            bool isInvalid = boundInterpolatedString.HasErrors;
+            SyntaxNode syntax = boundInterpolatedString.Syntax;
+            ITypeSymbol type = boundInterpolatedString.Type;
+            Optional<object> constantValue = ConvertToOptional(boundInterpolatedString.ConstantValue);
+            return new LazyInterpolatedStringExpression(parts, isInvalid, syntax, type, constantValue);
+        }
+
+        private static IInterpolatedStringContent CreateBoundInterpolatedStringContentOperation(BoundNode boundNode)
+        {
+            if (boundNode == null)
+            {
+                return null;
+            }
```

Comment:
Yes, the reason is a BoundLiteralExpression under a BoundInterpolatedExpressionString must map to an InterpolatedStringText operation node, not the common LiteralExpression node for strings seen at other places in the bound tree. Basically child bound nodes of an interpolated string are special. Do you see any other way to avoid this?